multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ProcessTimeCalculator.java

```diff
@@ -64,24 +63,46 @@ private long calculateFilteredProcessActivitiesTime(List<HistoricActivityInstanc
                                                         Predicate<HistoricActivityInstance> filter) {
         return processActivities.stream()
                                 .filter(filter)
-                                .mapToLong(HistoricActivityInstance::getDurationInMillis)
+                                .mapToLong(this::calculateActivityDuration)
                                 .sum();
     }
 
-    private boolean isCallActivity(HistoricActivityInstance historicActivityInstance) {
-        return CALL_ACTIVITY_TYPE.equals(historicActivityInstance.getActivityType());
+    // unfortunately, HistoricActivityInstance and HistoricProcessInstance have no common ancestor
```

Comment:
This will be obvious to anyone who tries to refactor this code, so this comment is unnecessary.

Reply:
Removed